week2-prac-todo/index.js

```diff
@@ -0,0 +1,219 @@
+import { initTodos } from "./data.js";
+
+// 요소 선택 !!!
+const input = document.querySelector('.todo-input');
+const prioritySelect = document.querySelector('.priority-input');
+const addBtn = document.querySelector('.add-btn');
+const todoList = document.querySelector('.todo-list');
+const dropdown = document.querySelector('.dropdown');
+const toggle = document.querySelector('.dropdown-toggle');
+const filterEntire = document.querySelector('#filter-button-entire');
+const filterCompleted = document.querySelector('#filter-button-completed');
+const filterUncompleted = document.querySelector('#filter-button-uncompleted');
+const filterPriority1 = document.querySelector('#filter-button-priority-1');
+const filterPriority2 = document.querySelector('#filter-button-priority-2');
+const filterPriority3 = document.querySelector('#filter-button-priority-3');
+const deleteBtn = document.querySelector('.delete-todo-button');
+const completeBtn = document.querySelector('.complete-todo-button');
+
+toggle.addEventListener('click', (e) => {
+  e.stopPropagation(); // 상위 이벤트 전파 방지
+  dropdown.classList.toggle('active');
+});
+
+document.addEventListener('click', () => {
+  dropdown.classList.remove('active');
+});
+
+document.addEventListener('DOMContentLoaded', () => {
+  const checkAll = document.querySelector('#check-all');
+
+  checkAll.addEventListener('change', (e) => {
+    const isChecked = e.target.checked;
+
+    const rowCheckboxes = document.querySelectorAll('.todo-list input[type="checkbox"]');
+
+    rowCheckboxes.forEach(checkbox => {
+      checkbox.checked = isChecked;
+    });
+  });
+});
+
+
+// 로컬스토리지에서 가져오기
+const storedTodos = JSON.parse(localStorage.getItem('todos') || '[]');
+
+const todos = [
+  ...storedTodos,
+  ...initTodos.filter(todo =>
+    !storedTodos.some(stored => stored.id === todo.id)
+  )
+];
```

Comment:
`some()`을 사용해서 id 중복 여부를 체크하여 `initTodos`와 `storedTodos`를 합치는 로직 깔끔하네요..! 다만 가독성을 위해, filter 결과를 변수로 나누고 병합하면 더 읽기 쉬울 것 같아요!


```suggestion
const newTodos = initTodos.filter(todo =>
  !storedTodos.some(stored => stored.id === todo.id)
);

const todos = [...storedTodos, ...newTodos];

```

Reply:
훠얼씬 깔끔하네용..